src/checks/missing-terminal-state-errors.ts

```diff
@@ -10,14 +10,13 @@ export const missingTerminalStateErrors: AslChecker = (definition) => {
     // - state with Type: Fail
     const terminals: Record<string, number> = {};
     let fsmId = 1;
-    JSONPath<States[]>({json: definition, path: '$..[\'States\']'})
+    getStatesContainer(definition)
```

Comment:
I extracted this code to a helper function for better re-use among other checkers that want to traverse the states